pkg/util/util.go

```diff
@@ -254,6 +254,42 @@ func DeepEqualDeployment(d1 *appsv1.Deployment, d2 *appsv1.Deployment) (bool, er
 	return true, nil
 }
 
+// DeepEqualDaemonSet compares if two daemonsets are same.
+func DeepEqualDaemonSet(d1, d2 *appsv1.DaemonSet) (bool, error) {
```

Comment:
can we make a function DeepEqualPodTemplate and call it by DeepEqualDaemonset and DeepEqualDeployment to avoid duplicate code.